realm/realm-library/src/objectServer/java/io/realm/mongodb/User.java

```diff
@@ -280,6 +282,18 @@ public State getState() {
         throw new IllegalStateException("Unknown state: " + nativeState);
     }
 
+    /**
+     * Return the custom user data associated with the user in the Realm App.
+     * <p>
+     * The data is only refreshed when the user's access token is refreshed.
+     *
+     * @return The custom user data associated with the user.
+     */
+    public Document getCustomData() {
```

Comment:
Looks like we also need to expose `refreshCustomData()`:  https://github.com/realm/realm-object-store/blob/v10/src/sync/app.hpp#L238 ... and since that is a network request, it should probably be:

```
public Document getCustomData()
public Document refreshCustomData()
public RealmAsyncTask refreshCustomData(Result<Document> callback)
```

Reply:
Will do. Should all these methods actually be on `OsUser` instead and what are the rationale for doing one over the other?

Reply:
So far we are using `OsUser` for all API's found on `sync_user.hpp` in ObjectStore.  `refresh_custom_data` is there as well, so should probably be there.

The primary reason for the split is that the `Os<X>` normally implements the `NativeObject` interface which we don't want in the public API, but it also allows us greater design flexibility in terms of how we expose the OS API.

It does have the downside there many methods are just "pass-through" though.

Reply:
Moved direct OS wrappers to `OsSyncUser` and kept the additional logic (returning user custom data after actual refresh) in `User`.